tensorflow/compiler/xla/service/gpu/cudnn_conv_runner.cc

```diff
@@ -364,6 +444,19 @@ Status RunCudnnConv(const HloCustomCallInstruction* conv,
                       stream, options);
 }
 
+// CuDNN Convolutions all go through and are dispatched from this function.
+// Dispatching are based on three conditions: input data type, output data type,
+// and convolution kind. Although these three conditions are independent, not
+// all combinations are supported: convolutions with float, double, and half
+// input must have the same output type and support all kinds of convolutions;
+// convolutions with int8 input allow both int8 and float output type, but only
+// support two kinds: kForward and kForwardActivation.
+//
+// This function itself dispatches convolutions for input and output data types;
+// RunCudnnConvInternalImpl later dispatches for convolution kind.
+// RunCudnnConvInternalImpl has two template specializations for floating point
+// types and for integer types, respectively, and only invoke the supported
+// convolutions.
```

Comment:
Maybe move the documentation about "data type dispatch" into a better place. I suggest to put it above one of the enable_ifs, and for the other one, simply say "see above ...".